apis/nucleus/src/components/listbox/components/ListBoxSearch.jsx

```diff
@@ -113,31 +113,48 @@ export default function ListBoxSearch({
     return response;
   };
 
+  function focusRow(container) {
+    const row = container?.querySelector('.last-focused') || container?.querySelector('[role="row"]:first-child');
+    row.setAttribute('tabIndex', 0);
+    row?.focus();
+  }
+
   const onKeyDown = async (e) => {
-    let response;
+    const { currentTarget } = e;
+    const container = currentTarget.closest('.listbox-container');
     switch (e.key) {
       case 'Enter':
-        response = await performSearch();
-        break;
-      case 'Escape':
-        response = cancel();
-        e.preventDefault();
-        e.stopPropagation();
+        return performSearch();
+      case 'Escape': {
+        focusRow(container);
+        return cancel();
+      }
+      case 'Tab': {
+        if (e.shiftKey) {
+          keyboard.focusSelection();
+        } else {
+          // Focus the row we last visited or the first one.
+          focusRow(container);
+
+          // Clean up.
+          container?.querySelectorAll('.last-focused').forEach((elm) => {
+            elm.classList.remove('last-focused');
+          });
+        }
         break;
+      }
       case 'f':
       case 'F':
         if (e.ctrlKey || e.metaKey) {
           if (hide) {
             hide();
           }
-          e.preventDefault();
-          e.stopPropagation();
```

Comment:
restore they are still needed for ctrl+F

Reply:
I put it back.

Reply:
ctrl+f - should call e.preventDefault();
f - should **not** call e.preventDefault();
it still looks that they do the same thing (currently not calling e.preventDefault())